sway-core/src/semantic_analysis/ast_node/expression/intrinsic_function.rs

```diff
@@ -240,6 +240,97 @@ impl TypedIntrinsicFunctionKind {
                     insert_type(TypeInfo::Boolean),
                 )
             }
+            Intrinsic::Gtf => {
+                if arguments.len() != 2 {
+                    errors.push(CompileError::IntrinsicIncorrectNumArgs {
+                        name: kind.to_string(),
+                        expected: 2,
+                        span,
+                    });
+                    return err(warnings, errors);
+                }
+
+                if type_arguments.len() != 1 {
+                    errors.push(CompileError::IntrinsicIncorrectNumTArgs {
+                        name: kind.to_string(),
+                        expected: 1,
+                        span,
+                    });
+                    return err(warnings, errors);
+                }
+
+                // Type check the first argument which is the index
+                let mut ctx = ctx
+                    .by_ref()
+                    .with_type_annotation(insert_type(TypeInfo::Unknown));
+                let index = check!(
+                    TypedExpression::type_check(ctx.by_ref(), arguments[0].clone()),
+                    return err(warnings, errors),
+                    warnings,
+                    errors
+                );
+
+                // Type check the second argument which is the tx field ID
+                let mut ctx = ctx
+                    .by_ref()
+                    .with_type_annotation(insert_type(TypeInfo::Unknown));
+                let tx_field_id = check!(
+                    TypedExpression::type_check(ctx.by_ref(), arguments[1].clone()),
+                    return err(warnings, errors),
+                    warnings,
+                    errors
+                );
+
+                // Make sure that the index argument is an unsigned integer
+                if !matches!(
+                    resolve_type(index.return_type, &index.span).unwrap(),
+                    TypeInfo::UnsignedInteger(_)
+                ) {
+                    errors.push(CompileError::IntrinsicUnsupportedArgType {
+                        name: kind.to_string(),
+                        span: index.span.clone(),
+                    });
+                }
+
+                // Make sure that the tx field ID is an unsigned integer
```

Comment:
So this (being able to pass consts to `__gtf`) is dependent on #810 then?

Reply:
Because we're not using `asm` block directly, the intrinsic will work with anything that is const-evaluatable like `const` declarations. So this will work:
```rust
    let u64_field = __gtf::<u64>(1, SOME_TX_FIELD);
    let b256_field = __gtf::<b256>(2, SOME_OTHER_TX_FIELD);
```
I added a small test to show this. The test compiles and the assembly has the right opcodes, but the test reverts as expected because the `fuel-core` changes are not done yet.

Reply:
A few things: 
- I think I'll close my `gtf` opcode PR in favor of this one because my test doesn't even compile yet.
>intrinsic will work with anything that is const-evaluatable like const declarations. 

great!

>it is possible that the gtf instruction gives us back a pointer to a b256

Yeah, I had been using `read` for the fields that return a pointer to a `b256` so this makes sense.